src/snyk/common/user.ts

```diff
@@ -0,0 +1,46 @@
+import { v4 as uuidv4 } from 'uuid';
+import { ISnykApiClient } from './api/apiСlient';
+import { MEMENTO_ANONYMOUS_ID } from './constants/globalState';
+import { ExtensionContext } from './vscode/extensionContext';
+
+export type UserDto = {
+  id: string;
+  username: string;
+};
+
+export class User {
+  private _authenticatedId?: string;
+
+  readonly anonymousId: string;
+
+  constructor(anonymousId?: string, authenticatedId?: string) {
+    this.anonymousId = anonymousId ?? uuidv4();
+    this._authenticatedId = authenticatedId ?? undefined;
+  }
+
+  static async get(context: ExtensionContext): Promise<User> {
+    let anonymousId = context.getGlobalStateValue<string>(MEMENTO_ANONYMOUS_ID);
+    if (!anonymousId) {
+      anonymousId = uuidv4();
+      await context.updateGlobalStateValue(MEMENTO_ANONYMOUS_ID, anonymousId);
+    }
+
+    return new User(anonymousId);
+  }
+
+  get authenticatedId(): string | undefined {
+    return this._authenticatedId;
+  }
+
+  async identify(apiClient: ISnykApiClient): Promise<void> {
```

Comment:
I would prefer the api client to be injected during construction, not on method call. Nothing blocking, though.

Reply:
The reason for method injection is that the uninstall script (`uninstall.ts`) would require providing full IConfiguration and ISnykApiClient objects to instantiate a class with the instance methods not being used at all. Providing it via method injection simplifies the external interface a lot.